File: backend/app/services/knowledge_base.py
"""知识库服务"""
from app.models.knowledge_base import KnowledgeBase, KnowledgeBaseMember, MemberRole, Visibility
from app.models.user import User


def create_knowledge_base(db, owner: User, name: str, description: str = "", visibility: Visibility = Visibility.PRIVATE) -> KnowledgeBase:
    """创建知识库"""
    kb = KnowledgeBase(
        name=name,
        description=description,
        visibility=visibility,
        owner_id=owner.id,
    )
    db.add(kb)
    db.commit()
    db.refresh(kb)
    return kb


def add_member(db, kb: KnowledgeBase, user_id: int, role: MemberRole) -> KnowledgeBaseMember:
    """添加成员"""
    member = KnowledgeBaseMember(
        knowledge_base_id=kb.id,
        user_id=user_id,
        role=role,
    )
    db.add(member)
    db.commit()
    db.refresh(member)
    return member


def update_member_role(db, member: KnowledgeBaseMember, role: MemberRole) -> KnowledgeBaseMember:
    """更新成员角色"""
    member.role = role
    db.commit()
    db.refresh(member)
    return member


def remove_member(db, member: KnowledgeBaseMember) -> None:
    """移除成员"""
    db.delete(member)
    db.commit()

File: frontend/src/api/knowledgeBase.ts
import client from "./client";

export type Visibility = "public" | "private";
export type MemberRole = "owner" | "admin" | "write" | "read";

export interface KnowledgeBase {
  id: number;
  name: string;
  description: string;
  visibility: Visibility;
  owner_id: number;
  owner_username?: string;
  created_at?: string;
  document_count: number;
}

export interface Member {
  id: number;
  user_id: number;
  username: string;
  email: string;
  role: MemberRole;
  created_at?: string;
}

export interface Document {
  id: number;
  filename: string;
  content_type: string;
  file_size: number;
  chunk_count: number;
  created_at?: string;
}

export const kbApi = {
  list: () => client.get<KnowledgeBase[]>("/knowledge-bases"),
  create: (data: { name: string; description?: string; visibility?: Visibility }) =>
    client.post<KnowledgeBase>("/knowledge-bases", data),
  get: (id: number) => client.get<KnowledgeBase>(`/knowledge-bases/${id}`),
  update: (id: number, data: { name?: string; description?: string; visibility?: Visibility }) =>
    client.patch<KnowledgeBase>(`/knowledge-bases/${id}`, data),
  delete: (id: number) => client.delete(`/knowledge-bases/${id}`),
  listDocuments: (id: number) => client.get<Document[]>(`/knowledge-bases/${id}/documents`),
  uploadDocument: (id: number, file: File) => {
    const form = new FormData();
    form.append("file", file);
    return client.post<{ document_id: number; chunk_count: number }>(
      `/knowledge-bases/${id}/documents`,
      form,
      { headers: { "Content-Type": "multipart/form-data" } }
    );
  },
  listMembers: (id: number) => client.get<Member[]>(`/knowledge-bases/${id}/members`),
  addMember: (id: number, data: { user_id: number; role: MemberRole }) =>
    client.post<Member>(`/knowledge-bases/${id}/members`, data),
  updateMember: (id: number, userId: number, role: MemberRole) =>
    client.patch<Member>(`/knowledge-bases/${id}/members/${userId}`, { role }),
  removeMember: (id: number, userId: number) =>
    client.delete(`/knowledge-bases/${id}/members/${userId}`),
};
